### PR TITLE
Add 'none' matcher for CLM module filters (bsc#1206817)

### PR DESCRIPTION
**Pull request by cbbayburt**
In RHEL 9, the default streams of modules are now included as regular packages instead of modules. To allow users to set up CLM with all the default streams, we must allow creating of regular repositories without having to add any module filters.

This PR adds a new matcher type `none` to AppStream filters. When used, it strips the modular metadata from the channel without having to specify any modules.

See also: [Documentation PR (#2007)](https://github.com/uyuni-project/uyuni-docs/pull/2007)

## GUI diff

Before:
Screenshot from 2023-01-17 20-43-31

After:
Screenshot from 2023-01-17 20-41-07
Screenshot from 2023-01-17 20-41-25

## Documentation
- [Documentation PR (#2007)](https://github.com/uyuni-project/uyuni-docs/pull/2007)

## Test coverage
- Unit tests were added

## Links

Fixes https://github.com/SUSE/spacewalk/issues/20032

## Changelogs

Make sure the changelogs entries you are adding are compliant with https://github.com/uyuni-project/uyuni/wiki/Contributing#changelogs and https://github.com/uyuni-project/uyuni/wiki/Contributing#uyuni-projectuyuni-repository

If you don't need a changelog check, please mark this checkbox:

- [ ] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

### CODE REVIEW

**Review by Etheryte — APPROVED** [review]
The frontend part looks good! Also good job on adding more tests, always nice when that improves in the frontend code. 👍

File: java/code/src/com/redhat/rhn/domain/contentmgmt/modulemd/ModularityDisabledException.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright (c) 2022 SUSE LLC
```
Comment (mcalmer) [inline java/code/src/com/redhat/rhn/domain/contentmgmt/modulemd/ModularityDisabledException.java:2]:
```suggestion
 * Copyright (c) 2023 SUSE LLC
```

**Review by mcalmer — APPROVED** [review]
One :nitpick only, the rest looks good

File: java/code/src/com/redhat/rhn/domain/contentmgmt/modulemd/ModularityDisabledException.java (hunk 1 above)
Comment (cbbayburt) [reply java/code/src/com/redhat/rhn/domain/contentmgmt/modulemd/ModularityDisabledException.java]: Done. Updated the checkstyle rule too.

**Review by parlt91 — APPROVED** [review]
Looks good to me. Thanks for being quite generous with adding new tests :smiley:

**Pull request merged**